docs/normalizer.md

```diff
@@ -1,40 +1,366 @@
+   'body' => 'That\'s amazing, everyone wanted a documentation of this tools package, and now this is done. All the team worked very hard to reach this goal, and i would personally thanks my mom who learned me to always push myself off limits !',
+   'category' => 'News'
+);
+
+var_export($article->normalize('full'));
+
+// will output
+array(
+   'title' => 'MajoraFrameworkExtraBundle documentation released',
+   'headline' => 'More than a year after release, MajoraFramework got a proper documentation !',
+   'body' => 'That\'s amazing, everyone wanted a documentation of this tools package, and now this is done. All the team worked very hard to reach this goal, and i would personally thanks my mom who learned me to always push myself off limits !',
+   'category' => array(
+        'name' => 'News',
+        'website' => 'www.github.com/LinkValue/MajoraFrameworkExtraBundle'
+   )
+);
 ```
-     namespace Alg\Identity\Component\Entity;
 
-     use Majora\Framework\Serializer\Model\SerializableTrait;
+### Denormalization
+
+Object denormalization follow the same logic : an interface behavior, implemented into a Php trait.
+
+On the other hand, denormalization doesn't use `getScope()` method, but mutators type hinting.
+Why implementing a denormalization by this way ? Because an object member which is not accessible throught a mutator shouldn't be accessible throught denormalization.
+
+With the same object structure :
+```php
+class Article implements NormalizableInterface
+{
+    use NormalizableTrait;
+
+    protected $title;
+    protected $category;
+
+    public function setTitle($title)
+    {
+        $this->title = $title;
+
+        return $this;
+    }
+
+    public function setCategory(Category $category)
+    {
+        $this->category = $category;
+
+        return $this;
+    }
+}
+
+class Category implements NormalizableInterface
+{
+    use NormalizableTrait;
+
+    protected $name;
+
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+}
+```
+With this call,
+```php
+$article = new Article();
+$article->denormalize(array(
+    'title' => 'MajoraFrameworkExtraBundle documentation released',
+    'category' => array(
+        'name' => 'News'
+    )
+));
+```
+normalizer will set scalar directly with PropertyAccessor component, and recurse on a "Category" object denormalization, because mutator defined a hinting on this class.
+
+If you don't have an instanciated object, you can use directly the normalizer class, which can create it :
+```php
+$article = MajoraNormalizer::createNormalizer()->denormalize(
+    array(
+        'title' => 'MajoraFrameworkExtraBundle documentation released',
+        'category' => array(
+            'name' => 'News'
+        )
+    ),
+    Article::class
+);
+```
+
+This is the proper way to denormalize entity using constructor parameters (Value objects / DDD), because Majora Normalizer detects construct parameters as well.
```

Comment:
Because Majora Normalizer detects construct parameters as well, this is the proper way to denormalize entity using constructor parameters (Value objects / DDD),